### PR TITLE
Show config under app query

### PR DESCRIPTION
**Pull request by stevengeeky**
aka issue #10

### CODE REVIEW

**Comment by stevengeeky** [comment]
Made it less complicated

File: util.js [hunk 1]
```diff
@@ -1054,31 +1054,27 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
...
         // prepare config to submit the app
-        let flattenedConfig = flattenConfig(app.config, []);
-        let flattenedUserConfig = flattenConfig(opt.config, []);
         let values = {};
         
-        Object.keys(flattenedConfig).forEach(key => {
-            if (flattenedConfig[key].type != 'input') {
-                let niceLookingKey = JSON.parse(key).join('.');
-                
+        Object.keys(app.config).forEach(key => {
```
Comment (soichih) [inline util.js]:
```
for(let key in app.config) {
...
}
```

??

File: util.js [hunk 2]
```diff
@@ -1054,31 +1054,27 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
...
                     } else {
-                        return reject("Error: no config entry found for key'" + niceLookingKey + "' (type: " + (flattenedConfig[key].type) + "). Please provide one and rerun");
+                        return reject("Error: no config entry found for key'" + key + "' (type: " + (app.config[key].type) + "). Please provide one and rerun");
                     }
                 }
 
-                if (flattenedUserConfig[key] && /boolean|string|number/.test(flattenedConfig[key].type)) {
-                    if (typeof flattenedUserConfig[key] != flattenedConfig[key].type) {
-                        return reject("Error: config key '" + niceLookingKey + "': expected type '" + flattenedConfig[key].type + "' but given value of type '" + (typeof flattenedUserConfig[key]) + "'");
+                if (opt.config[key] && /boolean|string|number/.test(app.config[key].type)) {
```
Comment (soichih) [inline util.js]: We've established  `if (app.config[key].type != 'input')` above, so we shouldn't have to check for the type here?

Just do
```
if(opt.config[key] === undefined)
```

"=== undefined" might be not necessary here if opt.config[key] is never set to false?

Maybe we should also check for enum type?

File: util.js [hunk 3]
```diff
@@ -1054,31 +1054,27 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
...
+                    if (typeof opt.config[key] != app.config[key].type) {
+                        return reject("Error: config key '" + key + "': expected type '" + app.config[key].type + "' but given value of type '" + (typeof opt.config[key]) + "'");
                     }
                 }
 
-                values[key] = flattenedUserConfig[key] || flattenedConfig[key].default;
+                values[key] = opt.config[key] || app.config[key].default;
```
Comment (soichih) [inline util.js]: Can use set `opt.config[key]` to false for boolean? Or 0 for number? If so, we might need to 
opt.config[key] === undefined || app.config[key].default.

`app.config[key]` is used a lot here.. I think it warrants  `let config = app.config[key]` at the top of for loop?

File: util.js [hunk 4]
```diff
@@ -1145,7 +1141,8 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
                 if (!opt.raw) console.log("Data Staging Task Created (" + body.task._id + ")");
                 
                 let task = body.task;
-                let preparedConfig = expandFlattenedConfig(flattenedConfig, values, task, inputs, datatypeTable, app);
+                let preparedConfig = prepareConfig(values, task, inputs, datatypeTable, app);
+                console.log(preparedConfig);
```
Comment (soichih) [inline util.js]: debug?

File: util.js [hunk 5]
```diff
@@ -1054,31 +1054,31 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
...
-                let niceLookingKey = JSON.parse(key).join('.');
-                
+        Object.keys(app.config).forEach(key => {
+            if (app.config[key].type != 'input') {
                 // validate each user-given config parameter
-                if (!flattenedUserConfig[key]) {
-                    if (flattenedConfig[key].default) {
-                        if (!opt.raw) console.log("No config entry found for key '" + niceLookingKey +
-                                    "'; using the default value in the app's config: " + flattenedConfig[key].default);
+                if (!opt.config[key]) {
```
Comment (soichih) [inline util.js]: Is this safe in case user set it to false, or 0?

File: util.js [hunk 6]
```diff
@@ -1054,31 +1054,31 @@ function runApp(headers, opt) {//appSearch, userInputs, projectSearch, resourceS
...
+                        return reject("Error: no config entry found for key'" + key + "' (type: " + (app.config[key].type) + "). Please provide one and rerun");
                     }
                 }
-
-                if (flattenedUserConfig[key] && /boolean|string|number/.test(flattenedConfig[key].type)) {
-                    if (typeof flattenedUserConfig[key] != flattenedConfig[key].type) {
-                        return reject("Error: config key '" + niceLookingKey + "': expected type '" + flattenedConfig[key].type + "' but given value of type '" + (typeof flattenedUserConfig[key]) + "'");
-                    }
+                
+                switch (app.config[key].type) {
```
Comment (soichih) [inline util.js]: Thanks for switching to switch. Now that look at it, should we add validator for enum type?

File: bl-app-query.js [hunk 7]
```diff
@@ -71,7 +71,7 @@ function formatApps(headers, data, whatToShow) {
                 
                 return formattedDatatype;
             }).join(', ');
-
+            
             let formattedOutputs = D.outputs.map(output => {
```
Comment (soichih) [inline bl-app-query.js]: Unrelated to this PR, but should "D" be renamed to "app"?

Comment (stevengeeky) [reply bl-app-query.js]: renamed (also renamed in other query files)

File: util.js (hunk 6 above)
Comment (stevengeeky) [reply util.js]: added in commit 71be59b

File: util.js (hunk 5 above)
Comment (stevengeeky) [reply util.js]: added explicit checking for undefined

File: util.js (hunk 4 above)
Comment (stevengeeky) [reply util.js]: removed

File: util.js (hunk 2 above)
Comment (stevengeeky) [reply util.js]: this code was changed

File: util.js (hunk 1 above)
Comment (stevengeeky) [reply util.js]: changed

**Pull request merged**